Frameworks/CoreGraphics/CGPattern.mm

```diff
@@ -114,6 +125,11 @@ - (CGAffineTransform)getPatternTransform {
     return matrix;
 }
 
+- (void)drawViaCallback:(CGContextRef)context {
+    // issue the callback
+    callbacks.drawPattern(info, context);
```

Comment:
By adding this method, you're increasing the objective-c burden. Please just use pattern->callbacks in the C function that calls this.

It will make it easier to move to CppBase later.

In fact, can you move this to CppBase?